File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/banner/service/BannerService.java
package b1nd.dodam.domain.rds.banner.service;

import b1nd.dodam.domain.rds.banner.entity.Banner;
import b1nd.dodam.domain.rds.banner.enumeration.BannerStatus;
import b1nd.dodam.domain.rds.banner.exception.BannerNotFoundException;
import b1nd.dodam.domain.rds.banner.repository.BannerRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class BannerService {

    private final BannerRepository repository;

    public void save(Banner banner) {
        repository.save(banner);
    }

    public void deleteById(int id) {
        repository.deleteById(id);
    }

    public Banner getById(int id) {
        return repository.findById(id)
                .orElseThrow(BannerNotFoundException::new);
    }

    public List<Banner> getActivates() {
        return repository.findByStatusOrderByIdDesc(BannerStatus.ACTIVE);
    }

    public List<Banner> getAllOrderByIdDesc() {
        return repository.findAllByOrderByIdDesc();
    }

}

File: services/service-banner/src/main/kotlin/com/b1nd/dodamdodam/banner/domain/banner/service/BannerService.kt
package com.b1nd.dodamdodam.banner.domain.banner.service

import com.b1nd.dodamdodam.banner.domain.banner.entity.BannerEntity
import com.b1nd.dodamdodam.banner.domain.banner.enumeration.BannerStatus
import com.b1nd.dodamdodam.banner.domain.banner.exception.BannerNotFoundException
import com.b1nd.dodamdodam.banner.domain.banner.repository.BannerRepository
import org.springframework.stereotype.Service

@Service
class BannerService(
    private val bannerRepository: BannerRepository,
) {
    fun create(banner: BannerEntity): BannerEntity = bannerRepository.save(banner)

    fun findById(id: Long): BannerEntity =
        bannerRepository.findById(id)
            .orElseThrow { BannerNotFoundException() }

    fun findAll(): List<BannerEntity> = bannerRepository.findAllByOrderByIdDesc()

    fun findActiveBanners(): List<BannerEntity> =
        bannerRepository.findByStatusOrderByIdDesc(BannerStatus.ACTIVE)

    fun deleteById(id: Long) {
        bannerRepository.deleteById(id)
    }
}
